### PR TITLE
[Minor] Improve arrow and parquet READMEs, document parquet feature flags

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Noticed while I was working on https://github.com/apache/arrow-rs/issues/2172

# Rationale for this change
 

The parquet crate has a non trivial number of feature flags. Let's consistently document the flags in crates.io and link to them in docs.rs

https://crates.io/crates/arrow
https://crates.io/crates/parquet

Also while I was messing around I found some other places where the docs could be improved so I did.

# What changes are included in this PR?
1. Document parquet feature flags
2. Misc wording improvements

# Are there any user-facing changes?
readme is different

### CODE REVIEW

File: arrow/README.md [hunk 1]
```diff
@@ -34,18 +37,24 @@ The arrow crate follows the [SemVer standard](https://doc.rust-lang.org/cargo/re
...
 - `ipc` (default) - support for the [arrow-flight](https://crates.io/crates/arrow-flight) IPC and wire format
 - `prettyprint` - support for formatting record batches as textual columns
 - `js` - support for building arrow for WebAssembly / JavaScript
-- `simd` - (_Requires Nightly Rust_) alternate optimized
+- `simd` - (_Requires Nightly Rust_) Use alternate hand optimized
   implementations of some [compute](https://github.com/apache/arrow-rs/tree/master/arrow/src/compute/kernels)
-  kernels using explicit SIMD instructions available through [packed_simd_2](https://docs.rs/packed_simd_2/latest/packed_simd_2/).
+  kernels using explicit SIMD instructions via [packed_simd_2](https://docs.rs/packed_simd_2/latest/packed_simd_2/).
 - `chrono-tz` - support of parsing timezone using [chrono-tz](https://docs.rs/chrono-tz/0.6.0/chrono_tz/)
+- `ffi` - bindings for the Arrow C [C Data Interface](https://arrow.apache.org/docs/format/CDataInterface.html)
```
Comment (alamb) [inline arrow/README.md:52]: newly added feature flags

File: arrow/src/lib.rs [hunk 2]
```diff
@@ -18,41 +18,8 @@
 //! A complete, safe, native Rust implementation of [Apache Arrow](https://arrow.apache.org), a cross-language
 //! development platform for in-memory data.
 //!
-//! # Performance Tips
```
Comment (alamb) [inline arrow/src/lib.rs:21]: I found this section was just introduced by @tustvold  in https://github.com/apache/arrow-rs/pull/2305 but it turns out there was already a section at the end of the README.md document that had the same information. I consolidated them in README.md and added a link "please see crates.io page" here

File: arrow/src/compute/README.md [hunk 3]
```diff
@@ -33,16 +33,16 @@ We use the term "kernel" to refer to particular general operation that contains
 
 Types of functions
 
-* Scalar functions: elementwise functions that perform scalar operations in a
+- Scalar functions: elementwise functions that perform scalar operations in a
```
Comment (alamb) [inline arrow/src/compute/README.md:36]: the `prettier` tool did this

File: arrow/README.md [hunk 4]
```diff
@@ -101,16 +110,38 @@ cargo run --example read_csv
 
 [arrow]: https://arrow.apache.org/
 
+## Performance Tips
 
-## Performance
+Arrow aims to be as fast as possible out of the box, whilst not compromising on safety. However,
```
Comment (alamb) [inline arrow/README.md:115]: This content was moved from what was added in https://github.com/apache/arrow-rs/pull/2305

File: parquet/README.md [hunk 5]
```diff
@@ -19,17 +19,38 @@
...
+
+The arrow crate follows the [SemVer standard](https://doc.rust-lang.org/cargo/reference/semver.html) defined by Cargo and works well within the Rust crate ecosystem.
+
+However, for historical reasons, this crate uses versions with major numbers greater than `0.x` (e.g. `19.0.0`), unlike many other crates in the Rust ecosystem which spend extended time releasing versions `0.x` to signal planned ongoing API changes. Minor arrow releases contain only compatible changes, while major releases may contain breaking API changes.
+
+## Feature Flags
+
+The `parquet` crate provides the following features which may be enabled in your `Cargo.toml`:
+
+- `arrow` (default) - support for reading / writing [`arrow`](https://crates.io/crates/arrow) arrays to / from parquet
```
Comment (alamb) [inline parquet/README.md:43]: here are the parquet feature flags I could find

File: arrow/README.md [hunk 6]
```diff
@@ -55,25 +64,25 @@ Arrow seeks to uphold the Rust Soundness Pledge as articulated eloquently [here]
 
 Where soundness in turn is defined as:
 
-> Code is unable to trigger undefined behaviour using safe APIs
+> Code is unable to trigger undefined behavior using safe APIs
```
Comment (tustvold) [inline arrow/README.md:67]: ... I shall British-ify the spelling one comment at a time 😂

**Review by tustvold — APPROVED** [review]

File: arrow/README.md [hunk 7]
```diff
@@ -101,16 +110,38 @@ cargo run --example read_csv
...
-## Performance
+Arrow aims to be as fast as possible out of the box, whilst not compromising on safety. However,
+it relies heavily on LLVM auto-vectorisation to achieve this. Unfortunately the LLVM defaults,
+particularly for x86_64, favour portability over performance, and LLVM will consequently avoid
+using more recent instructions that would result in errors on older CPUs.
 
-Most of the compute kernels benefit a lot from being optimized for a specific CPU target.
-This is especially so on x86-64 since without specifying a target the compiler can only assume support for SSE2 vector instructions.
-One of the following values as `-Ctarget-cpu=value` in `RUSTFLAGS` can therefore improve performance significantly:
+To address this it is recommended that you specify the override the LLVM defaults either
```
Comment (tustvold) [inline arrow/README.md:120]: Oops this keeps a typo from my PR, "specify the override the". Will fix

File: arrow/README.md (hunk 6 above)
Comment (alamb) [reply arrow/README.md:67]: lol I am "correcting" the spelling (or maybe I can claim I am "standardi**z**ing" the spelling 😆

**Pull request merged**